packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js

```diff
@@ -1473,6 +1473,38 @@ const Conversation = WebexPlugin.extend({
    * The value is one of 'reply', 'edit', 'cardAction', 'reaction', 'reactionSummary', 'reactionSelfSummary'
    */
 
+  /**
+   * Returns a list of _all_ child activities for a given parentId within a given conversation
+   * @param {object} [options = {}]
+   * @param {string} [options.conversationUrl] targeted conversation URL
+   * @param {string} [activityParentId] parent id of edit activities or thread activities
+   * @param {QueryOptions} [query] object containing query string values to be appended to the url
+   * @returns {Promise<Array>}
+   */
+  async listAllChildActivitiesByParentId(options = {}) {
+    const {conversationUrl, activityParentId, query} = options;
+
+    const initialResponse = await this.listChildActivitiesByParentId(conversationUrl, activityParentId, query);
+
+    let page = new Page(initialResponse, this.webex);
+
+    const items = [...page.items];
+
+    while (page.hasNext()) {
+      // eslint-disable-next-line no-await-in-loop
+      page = await page.next();
+      for (const activity of page) {
+        items.push(activity);
+      }
+    }
+
+    if (last(items).published < items[0].published) {
```

Comment:
What does this do? Could you add a comment as to why it is needed?

Reply:
I see in the tests that we could get them out of order. Should we do a sort on the array instead?

Reply:
Perhaps.. I wasn't exactly sure if it _was_ necessary, except that I found it in the _list code and thought that it might be a patch for unexpected server response ordering or something like that. It's simply checking if the first is newer than the last, and if so flipping the list. I'm not sure if it is needed at all.